test/src/100-publishstatistics/main

```diff
@@ -0,0 +1,243 @@
+  publish_repo $CVMFS_TEST_REPO > test_100_1 || return $?
+
+  # ============================ Test2 =============================
+
+  echo "init a new transaction to change something in repository $CVMFS_TEST_REPO"
+  start_transaction $CVMFS_TEST_REPO || return $?
+
+  echo "Test 2 - change stuff in the repository"
+  test2 $repo_dir || return 5
+  test2 $reference_dir || return 6
+
+  echo "creating CVMFS snapshot"
+  publish_repo $CVMFS_TEST_REPO > test_100_2 || return $?
+
+  # ============================ Test3 =============================
+
+  echo "Test 3 - making a tar and execute ingest command"
+  test3 $reference_dir || return 7
+  cvmfs_server ingest -t $reference_dir/$TAR_FILE -b $TAR_DIR $CVMFS_TEST_REPO > test_100_3 || return $?
+  cat test_100_3
+
+  # ============================ Test4 =============================
+
+  echo "Test 4 - overwrite the tar file"
+  cvmfs_server ingest -t $reference_dir/$TAR_FILE -b $TAR_DIR $CVMFS_TEST_REPO > test_100_4 || return $?
+  cat test_100_4
+
+  # ============================================================================
+
+  echo "check catalog and data integrity"
+  check_repository $CVMFS_TEST_REPO -i || return $?
+
+  # Reference test values
+  declare -A test_results
+  # test 1
+  test_results[1,1]=7
+  test_results[1,2]=0
+  test_results[1,3]=0
+  test_results[1,4]=2
+  test_results[1,5]=0
+  test_results[1,6]=0
+  test_results[1,7]=115
+  test_results[1,8]=0
+  # test 2
+  test_results[2,1]=0
+  test_results[2,2]=2
+  test_results[2,3]=2
+  test_results[2,4]=0
+  test_results[2,5]=1
+  test_results[2,6]=1
+  test_results[2,7]=31
+  test_results[2,8]=22
+  # test 3
+  test_results[3,1]=3
+  test_results[3,2]=0
+  test_results[3,3]=0
+  test_results[3,4]=2
+  test_results[3,5]=0
+  test_results[3,6]=0
+  test_results[3,7]=74
+  test_results[3,8]=0
+  # test 4
+  test_results[4,1]=0
+  test_results[4,2]=0
+  test_results[4,3]=3
+  test_results[4,4]=0
+  test_results[4,5]=0
+  test_results[4,6]=0
+  test_results[4,7]=0
+  test_results[4,8]=0
+
+  # check publish statistics
+  for i in `seq 1 $NR_OF_TESTS`;
+  do
+    local files_added=$(cat test_100_$i | grep "Files *added:" | awk '{print $NF}')
+    local files_removed=$(cat test_100_$i | grep "Files *removed:" | awk '{print $NF}')
+    local files_changed=$(cat test_100_$i | grep "Files *changed:" | awk '{print $NF}')
+    local directories_added=$(cat test_100_$i | grep "Directories *added:" | awk '{print $NF}')
+    local directories_removed=$(cat test_100_$i | grep "Directories *removed:" | awk '{print $NF}')
+    local directories_changed=$(cat test_100_$i | grep "Directories *changed:" | awk '{print $NF}')
+    local bytes_added=$(cat test_100_$i | grep "Bytes *added:" | awk '{print $NF}')
+    local bytes_removed=$(cat test_100_$i | grep "Bytes *removed:" | awk '{print $NF}')
+
+    if [ $files_added -eq ${test_results[$i,1]} ]; then
+      echo "Test $i.1 - number of added files --------- passed"
```

Comment:
I think we don't need to output the passed partial results, we can restrict the output to failed cases and save some lines of code.